elytron/WFLY-19269-OCSP-stapling-support.adoc

```diff
@@ -0,0 +1,199 @@
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+* Existing OCSP support for WildFly's Elytron subsystem would be extended to allow for OCSP stapling. A new attribute, called `enable-stapling`, would be added to `elytron/key-manager` to allow for stapling support. 
+
+** This attribute would be of type boolean. To allow for backwards compatibility, this attribute's value would be false by default, which would allow for traditional OCSP support. 
+
+Current Subsystem Configuration: 
+```
+<trust-manager name="trust-with-ocsp-crl" algorithm="PKIX" key-store="ElytronCaTruststore" soft-fail="true" only-leaf-cert="true" maximum-cert-path="2">
+    <certificate-revocation-list path="crl.pem"/>
+    <ocsp responder="responderUri" responder-certificate="certalias" responder-keystore="responder-store"/>
+</trust-manager>
+```
+
+Proposed Subsystem Configuration: 
+```
+<trust-manager name="trust-with-ocsp-crl" algorithm="PKIX" key-store="ElytronCaTruststore" soft-fail="true" only-leaf-cert="true" maximum-cert-path="2">
+    <certificate-revocation-list path="crl.pem"/>
+    <ocsp responder="responderUri" enable-stapling="true" responder-certificate="certalias" responder-keystore="responder-store"/>
+</trust-manager>
+```
+
+* This feature would make use of the existing `X509RevocationTrustManager` class to check the revocation status of the certificate.
+
+** This class makes use of the https://docs.oracle.com/javase/8/docs/api/java/security/cert/PKIXRevocationChecker.html[PKIXRevocationChecker] class, which checks the revocation status of the PKIX algorithm. 
+
+** This response is sent to the client using the `Certificate Status Request` extension response defined in https://datatracker.ietf.org/doc/html/rfc6961[RFC 6961]. 
+
+* The Elytron Client Configurations would be updated to also add a new element under the `ssl-context` tag to add the a configuration to enable OCSP stapling. 
+
+** This functionality would be the same as server configuration, and the default value for this would be set to `false`. 
+
+** Elytron Client Configuration schema would need to be bumped to version 1.8. 
+
+Current client configuration: 
+```
+<ssl-context name="ssl-crl-context">
+    <trust-store key-store-name="ca"/>
+    <trust-manager algorithm="PKIX" maximum-cert-path=4 soft-fail="true" only-leaf-cert="true"/>
+    <certificate-revocation-list path="crl.pem"/>
+    <ocsp responder="responderUri" responder-certificate="certalias" responder-keystore="responder-store"/>
+</ssl-context>
+```
+
+Proposed client configuration: 
+```
+<ssl-context name="ssl-crl-context">
+    <trust-store key-store-name="ca"/>
+    <trust-manager algorithm="PKIX" maximum-cert-path=4 soft-fail="true" only-leaf-cert="true"/>
+    <certificate-revocation-list path="crl.pem"/>
+    <ocsp responder="responderUri" enable-stapling=true responder-certificate="certalias" responder-keystore="responder-store"/>
```

Comment:
For the client configuration, the client needs to be able to make use of the OCSP response stapled to the certificate that's being presented by the server.

Reply:
New client config is TBD. 
I will look more into what needs to be defined and will update client side proposal.